commons-numbers-core/src/main/java/org/apache/commons/numbers/core/Multiplication.java

```diff
@@ -44,4 +44,12 @@ public interface Multiplication<T> {
      * @return <code>this<sup>-1</sup></code>.
      */
     T reciprocal();
+
+    /**
+     * Is this the neutral element of multiplication? Implementations may want to
+     * employ more efficient means than calling equals on the one element.
+     *
+     * @return {@code true} if {@code this} equals the result of {@link #one}.
```

Comment:
After the discussion about equals elsewhere, I am wondering what we really want to define here as the contract. I see two possibilities:
* true if this.equals(one())
* true if this is a representation of the neutral element of multiplication

Meanwhile you changed `DD.equals(double, double)` so the two may be the same now for the implementations available. From a usage perspective I would clearly hope they are always the same, but that would pose specific requirements for further implementations' `equals()` methods which may need to be documented somewhere, n particular given the precedent of BigDecimal's `equal()` not providing semantic equivalence but structural equivalence.

(Same discussion for `isZero()` of course.)

Reply:
BigDecimal.equals and Double.equals have chosen binary equality. A workaround for BigDecimal is to use compare.

We are of course free to do as we wish. I think the current documented behaviour of numerical equality is acceptable. The change I made to DD was to support this definition. So for now all 4 DD representations of zero are equal. All 2 representations of one are equal. The other classes used by Field also satisfy numerical equality. The FP64 has been changed so the two representations of zero are equal and all representations of NaN are equal.

Regarding the neutral element of multiplication, in some cases (e.g. DD) a multiplication by 1 may change the representation of the element if it contains signed zeros. But the magnitude will be unchanged. The output value will be equal to the input value when using the equals method.

If you wish to update the javadoc to reflect this behaviour then please do. E.g. state that this method returns true if multiplication by this element will not change the magnitude or signum of the multiplicand.

Reply:
Done